include/QuantumComputation.hpp

```diff
@@ -418,11 +446,14 @@ namespace qc {
             max_controls = std::max(ncontrols, max_controls);
         }
 
-        virtual VectorDD simulate(const VectorDD& in, std::unique_ptr<dd::Package>& dd) const;
-        virtual MatrixDD buildFunctionality(std::unique_ptr<dd::Package>& dd) const;
-        virtual MatrixDD buildFunctionalityRecursive(std::unique_ptr<dd::Package>& dd) const;
-        virtual bool     buildFunctionalityRecursive(std::size_t depth, std::size_t opIdx, std::stack<MatrixDD>& s, Permutation& permutation, std::unique_ptr<dd::Package>& dd) const;
+        virtual VectorDD                           simulate(const VectorDD& in, std::unique_ptr<dd::Package>& dd) const;
+        virtual std::map<std::string, std::size_t> simulate(const VectorDD& in, std::unique_ptr<dd::Package>& dd, std::size_t shots);
```

Comment:
Why is this method not const?

Reply:
This can't be `const` due to the function using the RNG and, consequently, modifying its state.